test/challtestsrv/dnsone.go

```diff
@@ -93,7 +93,26 @@ func (s *ChallSrv) dnsHandler(w dns.ResponseWriter, r *dns.Msg) {
 				m.Answer = append(m.Answer, record)
 			}
 		case dns.TypeCAA:
-			if q.Name == "bad-caa-reserved.com." || q.Name == "good-caa-reserved.com." {
+			addCAARecord := true
+
+			var value string
+			switch q.Name {
+			case "bad-caa-reserved.com.":
+				value = "sad-hacker-ca.invalid"
+			case "good-caa-reserved.com.":
+				value = "happy-hacker-ca.invalid"
+			case "recheck.good-caa-reserved.com.":
+				// Allow issuance when we're running in the past
+				// (under FAKECLOCK), otherwise deny issuance.
+				if os.Getenv("FAKECLOCK") != "" {
+					value = "happy-hacker-ca.invalid"
+				} else {
+					value = "sad-hacker-ca.invalid"
+				}
+			default:
+				addCAARecord = false
+			}
+			if addCAARecord {
```

Comment:
I think we could simplify this by just checking `value != ""`, right?

Reply:
While you could do this, it is technically valid for a CAA response to provide an empty value (which refuses issuance by any CA) - as such, I wanted to avoid conditioning on this.